frontend/ui/widget/progresswidget.lua

```diff
@@ -70,6 +70,8 @@ function ProgressWidget:paintTo(bb, x, y)
     }
     if self.dimen.w == 0 or self.dimen.h == 0 then return end
 
+    self._mirroredUI = self.fill_from_right or (BD.mirroredUILayout() and not self.fill_from_right)
```

Comment:
Is this just fine with the 2 other places (lines 90 and 125) that did use just `self._mirroredUI`  and not `self.fill_from_right` ?

If yes, you could make this a bit clearer to read by letting the global class constant line 55 (you need to restart KOReader when this changes), and here use:
```lua
if self.fill_from right then -- just do as if RTL UI
    self._mirroredUI = not self._mirroredUI
end
```
(Please check all works well with the Arabic mirrored UI)